tests/run-make/target-specs-filename/rmake.rs

```diff
@@ -0,0 +1,24 @@
+// This test checks the rustc can accept target-spec-json with cases
+// including file extension - it could be in upper and lower case.
+// Used to test: print target-spec-json and cfg
+// Issue: https://github.com/rust-lang/rust/issues/127387
+
+use run_make_support::rustc;
+
+fn main() {
+    // This is a matrix [files x args], files for target, args for print.
+    for file in [
+        "normal-target-triple",
+        "normal-target-triple.json",
+        "ext-in-caps.JSON",
+        "ALL-IN-CAPS.JSON",
+    ] {
+        for args in [["cfg"].as_slice(), &["target-spec-json", "-Zunstable-options"]] {
+            let output = rustc().arg("--print").args(args).args(["--target", file]).run();
```

Comment:
Suggestion: sorry I know there's currently no docs for the support library (still working on it), but `.run()` asserts that the output status is succecssful so you don't need the status assert ^^

Reply:
I've realized it couple minutes ago. Thanks, I'll remove it.

Reply:
Done ❤️‍🔥